### PR TITLE
impl SecurityDoor::VUpdate_461AD0

### PR DESCRIPTION
**Pull request by mouzedrift**
i didn't add the states enum yet because it looks a bit different compared to AE and i'm not really in the mood RN after VS intellisense decided to not work whatsoever.

### CODE REVIEW

File: Source/AliveLibAO/SecurityDoor.cpp [hunk 1]
```diff
@@ -99,5 +112,241 @@ void SecurityDoor::VScreenChanged_461F80()
...
+
+        if (last_event == GameSpeakEvents::eNone_m1)
+        {
+            const auto MatchBuffer = pEventSystem_4FF954->MatchBuffer_40FAA0(field_F8_stru, field_108_max_idx, field_10A_event_idx);
+            field_11A_unused = static_cast<short>(MatchBuffer);
+            if (MatchBuffer == GameSpeakMatch::eFullMatch_1 || sVoiceCheat_507708)
+            {
+                field_10_anim.field_4_flags.Clear(AnimFlags::eBit3_Render);
+                SwitchStates_Set(field_EA_id, 1);
+                SFX_Play_43AE60(112u, 127, -700, 0);
```
Comment (paulsapps) [inline Source/AliveLibAO/SecurityDoor.cpp:322]: forgot ya enum

File: Source/AliveLibAO/SecurityDoor.cpp [hunk 2]
```diff
@@ -99,5 +112,241 @@ void SecurityDoor::VScreenChanged_461F80()
...
+                field_E8_state = 1;
+                field_114_timer = gnFrameCount_507670 + 15;
+            }
+            else
+            {
+                SFX_Play_43AE60(45u, 60, -720, 0);
```
Comment (paulsapps) [inline Source/AliveLibAO/SecurityDoor.cpp:328]: ditto

File: Source/AliveLibAO/SecurityDoor.cpp [hunk 3]
```diff
@@ -99,5 +112,241 @@ void SecurityDoor::VScreenChanged_461F80()
...
+                field_114_timer = gnFrameCount_507670 + 15;
+            }
+        }
+        break;
+    }
+
+    case 10:
+        if (static_cast<int>(gnFrameCount_507670) > field_114_timer)
+        {
+            SFX_Play_43AE60(110u, 127, -1000, 0);
```
Comment (paulsapps) [inline Source/AliveLibAO/SecurityDoor.cpp:339]: ditto

File: Source/AliveLibAO/SecurityDoor.cpp [hunk 4]
```diff
@@ -99,5 +112,241 @@ void SecurityDoor::VScreenChanged_461F80()
...
+            {
+                field_10_anim.field_4_flags.Clear(AnimFlags::eBit3_Render);
+            }
+        }
+        break;
+
+    case 1:
+        if (static_cast<int>(gnFrameCount_507670) == field_114_timer)
+        {
+            SND_SEQ_Play_477760(45u, 1, 127, 127);
```
Comment (paulsapps) [inline Source/AliveLibAO/SecurityDoor.cpp:160]: enum

File: Source/AliveLibAO/SecurityDoor.cpp [hunk 5]
```diff
@@ -99,5 +112,241 @@ void SecurityDoor::VScreenChanged_461F80()
...
+    case 8:
+    {
+        if (static_cast<int>(gnFrameCount_507670) > field_114_timer)
+        {
+            SFX_Play_43AE60(SoundEffect::Alarm_45, 60, -720, 0);
+            field_E8_state = 10;
+            field_114_timer = gnFrameCount_507670 + 15;
+        }
+
+        GameSpeakEvents last_event;
```
Comment (paulsapps) [inline Source/AliveLibAO/SecurityDoor.cpp:269]: = {} ;

File: Source/AliveLibAO/SecurityDoor.cpp [hunk 6]
```diff
@@ -99,5 +112,241 @@ void SecurityDoor::VScreenChanged_461F80()
...
+    case 2:
+        Slig::Slig_GameSpeak_SFX_46F560(SligSpeak::eHi_0, 127, -200, 0);
+        field_10_anim.Set_Animation_Data_402A40(992, 0);
+        field_E8_state = 3;
+        field_114_timer = gnFrameCount_507670 + 150;
+        break;
+
+    case 3:
+    {
+        GameSpeakEvents last_event;
```
Comment (paulsapps) [inline Source/AliveLibAO/SecurityDoor.cpp:173]: = {};

File: Source/AliveLibAO/SecurityDoor.cpp (hunk 4 above)
Comment (mouzedrift) [reply Source/AliveLibAO/SecurityDoor.cpp:160]: those don't have enums in AO yet

Comment (paulsapps) [reply Source/AliveLibAO/SecurityDoor.cpp:160]: Could add a place holder so the rename catches it?

**Pull request merged**